src/pages/index.js

```diff
@@ -74,6 +134,55 @@ function handleImageClick(cardData) {
   imagePopup.open(cardData);
 }
 
+const deleteCardPopup = new PopupWithConfirm("#delete-card-modal", () => {});
+deleteCardPopup.setEventListeners();
+
+//runs when you click on a card's delete button
+function handleDeleteCard(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitHandler(handleDeleteCardFormSubmit);
+
+  //runs when you click the 'yes' button on the delete card modal
+  function handleDeleteCardFormSubmit() {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.handleDeleteClick();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+        alert("Could not delete place!");
+      })
+      .finally(() => {
+        yesConfirmationButton.disabled = false;
+        yesConfirmationButton.classList.remove("modal__button_disabled");
```

Comment:
Now only `FormValidator` should **`disable/enable`** submit buttons and remove errors from the inputs (if you do it). It’s a violation of **Single responsibility** principle